rust/agents/scraper/migration/src/m20220805_000005_create_table_message_state.rs

```diff
@@ -0,0 +1,95 @@
+use sea_orm_migration::prelude::*;
+
+use crate::l20220805_types::*;
+use crate::m20220805_000004_create_table_message::Message;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .create_table(
+                Table::create()
+                    .table(MessageState::Table)
+                    .if_not_exists()
+                    .col(
+                        ColumnDef::new(MessageState::Id)
+                            .big_integer()
+                            .not_null()
+                            .auto_increment()
+                            .primary_key(),
+                    )
+                    .col(
+                        ColumnDef::new(MessageState::TimeCreated)
+                            .timestamp()
+                            .not_null(),
+                    )
+                    .col(ColumnDef::new(MessageState::MsgId).big_integer().not_null())
+                    .col(
+                        ColumnDef::new(MessageState::BlockHeight)
+                            .big_unsigned()
+                            .not_null(),
+                    )
+                    .col(
+                        ColumnDef::new(MessageState::BlockTimestamp)
+                            .timestamp()
+                            .not_null(),
+                    )
+                    .col(
+                        ColumnDef::new(MessageState::Processable)
+                            .boolean()
+                            .not_null(),
+                    )
+                    .col(&mut ColumnDef::new_with_type(
+                        MessageState::EstimatedGasCost,
+                        CryptoCurrency,
+                    ))
+                    .col(ColumnDef::new(MessageState::ErrorMsg).text())
+                    .foreign_key(
+                        ForeignKey::create()
+                            .from_col(MessageState::MsgId)
+                            .to(Message::Table, Message::Id),
+                    )
+                    .index(
+                        Index::create()
+                            .col(MessageState::MsgId)
+                            .col(MessageState::BlockHeight)
+                            .unique(),
+                    )
+                    .to_owned(),
+            )
+            .await
+    }
+
+    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .drop_table(Table::drop().table(MessageState::Table).to_owned())
+            .await
+    }
+}
+
+/// Learn more at https://docs.rs/sea-query#iden
+#[derive(Iden)]
+pub enum MessageState {
```

Comment:
Relating to my comment above, is there a reason we can't have these as nullable fields in the message table?

Reply:
Yes, see above.